src/devices.ts

```diff
@@ -53,7 +71,39 @@ export const createDevicesEndpoints = (
     });
 
     if (response.error) {
-      throw new Error(response.error.error);
+      if ((response.error as RawMissingFirmwareAssetError).missingFwAssets) {
+        if (process.env.FETCH_FIRMWARE_ASSETS === '1') {
+          const knownErr = new MissingFirmwareAssetError (response.error as RawMissingFirmwareAssetError)
+          for (const firmwareAssetUrl of knownErr.missingFwAssets) {
+            const response = await fetch(firmwareAssetUrl);
+            const fwAssetPath = path.join(os.tmpdir(), `${crypto.randomUUID()}.fwasset`)
```

Comment:
This could be inside the `response.ok` block as it's not necessary to create if the fetch failed